src/ast/codegen_llvm.cpp

```diff
@@ -1539,6 +1539,21 @@ void CodegenLLVM::visit(FieldAccess &acc)
       Value *masked = b_.CreateAnd(shifted, field.bitfield.mask);
       expr_ = masked;
     }
+    else if (field.type.IsIntTy() && field.is_data_loc)
+    {
+      // `is_data_loc` should only be set if field access is on `args` which
+      // has to be a ctx access
+      assert(type.IsCtxAccess());
+      assert(ctx_->getType() == b_.getInt8PtrTy());
+
+      // Top 2 bytes are length (which we'll ignore). Bottom two bytes are
+      // offset which we add to the start of the tracepoint struct.
+      expr_ = b_.CreateLoad(b_.getInt32Ty(),
```

Comment:
Loading an i32 from an i8* here which makes llvm complain. Inserting a  `b_.CreatePointerCast(ctx_, b_.getInt32Ty()->getPointerTo()` shoud fix it

Reply:
Oops, comment should go on the line below.

Also requires `field.offset/4` as creategep will now work with 32b steps.

#1396 will help clean this up

Reply:
Fixed